### PR TITLE
Fix TaskState after error in Task::start()

### PR DESCRIPTION
**Pull request by oerling**
Catches errors in constructing Drivers and sete the Task's state and error accordingly.

### CODE REVIEW

**Comment by spershin** [comment]
@mbasmanova 
In Task::start() not only the Driver creation can throw, but pretty much any code.
Should we wrap the whole function in the try?

File: velox/exec/Task.cpp [hunk 1]
```diff
@@ -553,71 +553,78 @@ void Task::start(
     }
   }
 
-  std::unique_lock<std::mutex> l(self->mutex_);
```
Comment (xiaoxmeng) [inline velox/exec/Task.cpp:556]: Let's keep this under task lock?

File: velox/exec/Task.cpp [hunk 2]
```diff
@@ -553,71 +553,78 @@ void Task::start(
...
+    // As some splits for grouped execution could have been added before the
+    // task start, ensure we start running drivers for them.
+    if (self->numDriversPerSplitGroup_ > 0) {
+      if (!l.owns_lock()) {
+        l.lock();
+      }
+      self->ensureSplitGroupsAreBeingProcessedLocked(self);
     }
-    self->ensureSplitGroupsAreBeingProcessedLocked(self);
+  } catch (const std::exception& e) {
```
Comment (xiaoxmeng) [inline velox/exec/Task.cpp:625]: Do we expect non velox exception here?

**Review by xiaoxmeng — COMMENTED** [review]
@oerling thanks for fix % minors!

**Pull request closed without merging**